src/wallet/wallet.h

```diff
@@ -653,17 +653,35 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
      */
     bool AddWatchOnly(const CScript& dest) override;
 
+    std::unique_ptr<CWalletDBWrapper> dbw;
+
 public:
     /*
      * Main wallet lock.
      * This lock protects all the fields added by CWallet
      *   except for:
      *      fFileBacked (immutable after instantiation)
-     *      strWalletFile (immutable after instantiation)
      */
     mutable CCriticalSection cs_wallet;
 
-    const std::string strWalletFile;
+    /** Get database handle used by this wallet. Ideally this function would
+     * not be necessary.
+     */
+    CWalletDBWrapper *GetDBHandle()
+    {
+        return dbw.get();
+    }
+
+    /** Get a name for this wallet for logging/debugging purposes.
+     */
+    std::string GetName()
+    {
+        if (dbw) {
+            return dbw->GetName();
+        } else {
```

Comment:
I prefer to keep it this way. We have no guideline on this in `doc/developer-notes.md`, and I personally find it easier to follow the control flow this way.